src/renderers/webgl/WebGLProgram.js

```diff
@@ -816,7 +826,8 @@ function WebGLProgram( renderer, cacheKey, parameters, bindingStates ) {
 
 		if ( cachedUniforms === undefined ) {
 
-			cachedUniforms = new WebGLUniforms( gl, program );
+			// Populates cachedUniforms and cachedAttributes
+			onFirstUse();
```

Comment:
On some devices here an issue appears [on some mac at least].
When calling this function inside this context is lost [this is just undefined].
This can be fixed with a hack [force context passing] doing onFirstUse.call( this ) 
and this actually fixes the issue.
Trace:
![Screenshot 2022-01-26 at 19 26 38](https://user-images.githubusercontent.com/1478866/151529657-777d7f68-4dd1-4a6a-9954-002457fb419c.png)

Reply:
And debugger screenshot:
![photo_2022-01-28 11 21 52](https://user-images.githubusercontent.com/1478866/151530163-2721c9e4-810e-43ce-94e0-98b300199c6e.jpeg)

Reply:
It's a bug in https://github.com/mrdoob/three.js/pull/19745 -- see https://github.com/mrdoob/three.js/pull/19745#discussion_r1031384089.